hashintel/hash
# Array property type creation

## 🌟 What is the purpose of this PR?

This PR focuses on implementing the logic for creating array property types.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1203301853897838/1203366629132085/f) _(internal)_

## 🔍 What does this change?

- Added menus for creating a custom property type (only arrays in this PR);
- Moved some of the form logic on `property-type-form` to `property-utils`;
- Moved the dropdown on `property-type-selector-dropdown` to a generic component so it can be reused.

## 🚫 Blocked by

- [x]  #1527 
- [x]  #1521

## 🐾 Next steps

- Fix Modal positioning;
- Finish styling;
- Other subtasks [here](https://app.asana.com/0/1203301853897838/1203343922163681/f).

## ❓ How to test this?

1.  Start creating an entity type;
2. Add a custom property type;
3. Add an array (and nested properties) to the custom property type;
4. Save the property type;
5. Check if it was correctly saved in the database.

## 📹 Demo
